src/main/java/bio/terra/service/snapshotbuilder/SnapshotRequestDao.java

```diff
@@ -0,0 +1,164 @@
+    } catch (EmptyResultDataAccessException ex) {
+      throw new NotFoundException("No snapshot access requests found for given id", ex);
+    }
+  }
+
+  /**
+   * Return the list of Snapshot Requests associated with the given dataset id.
+   *
+   * @param datasetId associated with any number of snapshot requests.
+   * @return the list of snapshot requests, empty if none, or an exception if the dataset does not
+   *     exist.
+   */
+  @Transactional(propagation = Propagation.REQUIRED, readOnly = true)
+  public List<SnapshotAccessRequestResponse> enumerateByDatasetId(UUID datasetId) {
+    String sql = "SELECT * FROM snapshot_request WHERE " + DATASET_ID + " = :dataset_id";
+    MapSqlParameterSource params = new MapSqlParameterSource().addValue(DATASET_ID, datasetId);
+    try {
+      return jdbcTemplate.query(sql, params, responseMapper);
+    } catch (EmptyResultDataAccessException ex) {
+      throw new NotFoundException("No snapshot requests found for given dataset id", ex);
+    }
+  }
+
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public SnapshotAccessRequestResponse create(
+      UUID datasetId, SnapshotAccessRequest request, String email) {
+    String jsonValue;
+    try {
+      jsonValue = objectMapper.writeValueAsString(request.getDatasetRequest());
+    } catch (JsonProcessingException e) {
+      throw new BadRequestException("Could not write snapshot access request to json", e);
+    }
+    DaoKeyHolder keyHolder = new DaoKeyHolder();
+    String sql =
+        """
+        INSERT INTO snapshot_request
+        (dataset_id, snapshot_name, snapshot_research_purpose, snapshot_specification, created_by)
+        VALUES (:dataset_id, :snapshot_name, :snapshot_research_purpose, cast(:snapshot_specification as jsonb), :created_by)
+        """;
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue(DATASET_ID, datasetId)
+            .addValue(SNAPSHOT_NAME, request.getName())
+            .addValue(SNAPSHOT_RESEARCH_PURPOSE, request.getResearchPurposeStatement())
+            .addValue(SNAPSHOT_SPECIFICATION, jsonValue)
+            .addValue(CREATED_BY, email);
+    try {
+      jdbcTemplate.update(sql, params, keyHolder);
+    } catch (DataIntegrityViolationException ex) {
+      throw new NotFoundException("Dataset with given dataset id does not exist.");
+    }
+    UUID id = keyHolder.getId();
+    return getById(id);
+  }
+
+  @Transactional(propagation = Propagation.REQUIRED, isolation = Isolation.SERIALIZABLE)
+  public SnapshotAccessRequestResponse update(
+      UUID requestId, SnapshotAccessRequestResponse.StatusEnum status) {
+    String sql =
+        "UPDATE snapshot_request SET "
+            + STATUS
+            + " = :status, "
+            + UPDATED_DATE
+            + "= :updated_date "
+            + "WHERE "
+            + ID
+            + " = :id";
```

Comment:
Not sure what the TDR convention is for this, but I would use a text block and `formatted()` to write this. Something like:
```suggestion
        """
        UPDATE snapshot_request SET %s = :status, %s = :updated_date WHERE %s = :id
        """.formatted(STATUS, UPDATED_DATE, ID);
```

Although I see that this is to share column names with sql text. I like the idea of sharing constants like this, but any time you build a sql query by hand you have to worry about sql injection. Personally I'd use the constants for any case outside of a SQL string and leave the SQL as an unformatted literal when possible. This lets the IDE scan the and check the SQL too, if you have the TDR database configured for your project.